keras/engine/training.py

```diff
@@ -2404,13 +2416,18 @@ def evaluate_generator(self, generator, steps=None,
                 enqueuer.stop()
 
         if not isinstance(outs, list):
+            assert not stateful_metric_indices
```

Comment:
Why? Please add a comment.

Reply:
In case of loss-only models, obviously no stateful_metric_indices may exist.

I have now instead refactored the code to be consistent with all other Model methods in handling of loss-only models.